src/metrics/list_repos.rs

```diff
@@ -32,26 +35,115 @@ impl ListReposForOrg {
     }
 }
 
+impl ListReposForOrg {
+    fn to_repo(&self, repo_name: &str) -> Repo {
+        Repo {
+            graphql: self.graphql.clone(),
+            org_name: self.org_name.clone(),
+            repo_name: repo_name.to_string(),
+            start_date: self.start_date.clone(),
+            end_date: self.end_date.clone(),
+        }
+    }
+}
+
 #[async_trait]
 impl Producer for ListReposForOrg {
     fn column_names(&self) -> Vec<String> {
-        vec![String::from("Repository Name"), String::from("# of PRs")]
+        vec![
+            String::from("Organization"),
+            String::from("Repository"),
+            String::from("PRs Opened"),
+            String::from("Issues Opened"),
+            String::from("Issues Closed"),
+            String::from("Start Date"),
+            String::from("End Date"),
+        ]
     }
 
     async fn producer_task(mut self, tx: Sender<Vec<String>>) -> Result<(), eyre::Error> {
-        for repo in &self.repo_names {
-            let count_prs = util::count_pull_requests(
-                &mut self.graphql,
-                &self.org_name,
-                &repo,
-                &self.start_date,
-                &self.end_date,
-            )
+        for repo_name in &self.repo_names {
+            let mut repo = self.to_repo(repo_name);
+            let count_prs = repo.count_pulls().await?;
+            let count_issues = repo.count_issue_closures().await?;
+
+            tx.send(vec![
+                self.org_name.clone(),
+                repo_name.to_owned(),
+                count_prs.to_string(),
+                count_issues.opened.to_string(),
+                count_issues.closed.to_string(),
+                self.start_date.to_string(),
+                self.end_date.to_string(),
+            ])
             .await?;
-            tx.send(vec![repo.to_owned(), count_prs.to_string()])
-                .await?;
         }
 
         Ok(())
     }
 }
+
+#[derive(Default, Debug)]
+struct IssueClosuresCount {
+    opened: usize,
+    closed: usize,
+}
+
+#[derive(Clone, Debug)]
+struct Repo {
+    graphql: Graphql,
+    org_name: String,
+    repo_name: String,
+    start_date: Datetime,
+    end_date: Datetime,
+}
+
+impl Repo {
+    #[throws]
+    async fn count_issue_closures(&self) -> IssueClosuresCount {
+        let mut repo = self.clone();
+        let mut clone = self.clone();
+
+        let futures = vec![repo.count_issues("created"), clone.count_issues("closed")];
```

Comment:
I don't want to delay this any more, but wouldn't [`try_join`](https://rust-lang-nursery.github.io/futures-api-docs/0.3.0-alpha.4/futures/future/trait.TryFutureExt.html#method.try_join) work?

Reply:
I do like that better 😄 . pushed up that change